src/mappings/ContributionRewardExt/mapping.ts

```diff
@@ -0,0 +1,187 @@
+
+export function handleRedeemReputation(event: RedeemReputation): void {
+  updateProposalAfterRedemption(event.address, event.params._proposalId, 0);
+  let ent = new ContributionRewardRedeemReputation(eventId(event));
+  ent.txHash = event.transaction.hash;
+  ent.contract = event.address;
+  ent.amount = event.params._amount;
+  ent.avatar = event.params._avatar;
+  ent.beneficiary = event.params._beneficiary;
+  ent.proposalId = event.params._proposalId;
+  store.set('ContributionRewardRedeemReputation', ent.id, ent);
+}
+
+export function handleRedeemNativeToken(event: RedeemNativeToken): void {
+  updateProposalAfterRedemption(event.address, event.params._proposalId, 1);
+  let ent = new ContributionRewardRedeemNativeToken(eventId(event));
+  ent.txHash = event.transaction.hash;
+  ent.contract = event.address;
+  ent.amount = event.params._amount;
+  ent.avatar = event.params._avatar;
+  ent.beneficiary = event.params._beneficiary;
+  ent.proposalId = event.params._proposalId;
+  store.set('ContributionRewardRedeemNativeToken', ent.id, ent);
+}
+
+export function handleRedeemEther(event: RedeemEther): void {
+  updateProposalAfterRedemption(event.address, event.params._proposalId, 2);
+  let ent = new ContributionRewardRedeemEther(eventId(event));
+  ent.txHash = event.transaction.hash;
+  ent.contract = event.address;
+  ent.amount = event.params._amount;
+  ent.avatar = event.params._avatar;
+  ent.beneficiary = event.params._beneficiary;
+  ent.proposalId = event.params._proposalId;
+  store.set('ContributionRewardRedeemEther', ent.id, ent);
+}
+
+export function handleRedeemExternalToken(event: RedeemExternalToken): void {
+  updateProposalAfterRedemption(event.address, event.params._proposalId, 3);
+  let ent = new ContributionRewardRedeemExternalToken(eventId(event));
+  ent.txHash = event.transaction.hash;
+  ent.contract = event.address;
+  ent.amount = event.params._amount;
+  ent.avatar = event.params._avatar;
+  ent.beneficiary = event.params._beneficiary;
+  ent.proposalId = event.params._proposalId;
+  store.set('ContributionRewardRedeemExternalToken', ent.id, ent);
+}
+
+function insertNewProposal(event: NewContributionProposal): void {
+  let ent = new ContributionRewardProposal(event.params._proposalId.toHex());
+  ent.proposalId = event.params._proposalId;
+  ent.contract = event.address;
+  ent.avatar = event.params._avatar;
+  ent.beneficiary = event.params._beneficiary;
+  ent.descriptionHash = event.params._descriptionHash;
+  ent.externalToken = event.params._externalToken;
+  ent.votingMachine = event.params._intVoteInterface;
+  ent.reputationReward = event.params._reputationChange;
+  let rewards = event.params._rewards;
+  ent.nativeTokenReward = rewards.shift(); // native tokens
+  ent.ethReward = rewards.shift(); // eth
+  ent.externalTokenReward = rewards.shift(); // external tokens
+  ent.periods = BigInt.fromI32(1); // number of periods
+  store.set('ContributionRewardProposal', ent.id, ent);
```

Comment:
ContributionReward and ContributionRewardExt are different schemes - are you sure you want to use the same entitites here?

For example, the Ext version has fields like `reputationRewardLeft`, where are these going to be stored?

Reply:
I think so, they're essentially the same, and this is the approach we took with the U/GenericScheme.